web/auth/jwt.go

```diff
@@ -56,39 +55,35 @@ func (tm *TokenManager) JWTUpdateClaims(db database.Database, userID uint64) (*C
 	return newClaims, nil
 }
 
+// TODO(vera): needs access to the database to update changes there
 // Update will remove a user ID at the index from the list
-func (tm *TokenManager) Remove(index uint64) {
+func (tm *TokenManager) Remove(index uint64, db database.Database) {
 	updatedTokenList := append(tm.TokensToUpdate[:index], tm.TokensToUpdate[index+1:]...)
 	tm.TokensToUpdate = updatedTokenList
 }
 
 // Add adds a new UserID to the list of users who require an updated JWT token
-func (tm *TokenManager) Add(userID uint64) {
+func (tm *TokenManager) Add(userID uint64, logger *zap.SugaredLogger, db database.Database) {
 	// Return if the given ID is already in the list to avoid duplicates
-	for _, id := range tm.TokensToUpdate {
-		if userID == id {
+	for _, token := range tm.TokensToUpdate {
+		if userID == token.UserID {
 			return
 		}
 	}
-	tm.TokensToUpdate = append(tm.TokensToUpdate, userID)
-}
-
-// String creates a string of all IDs in the list to store in the database
-func (tm *TokenManager) String() string {
-	return strings.Trim(strings.Join(strings.Split(fmt.Sprint(tm.TokensToUpdate), " "), ","), "[]")
+	tokenQuery := &pb.UpdateTokenRecord{UserID: userID}
+	if err := db.CreateTokenRecord(tokenQuery); err != nil {
+		logger.Errorf("error updating token record in the database: %w", err)
+	}
+	// TODO(vera): remove, testing if ID is getting updated, must be tested in the corresponding test
+	logger.Debugf("Token query after update in the database: %+v", tokenQuery)
+	tm.TokensToUpdate = append(tm.TokensToUpdate, tokenQuery)
 }
 
 // UpdateFromDatabase splits string of user IDs stored in the database into a list
-func (tm *TokenManager) UpdateFromDatabase(stringWithIDs string) {
-	stringIDs := strings.Split(stringWithIDs, ", ")
-	integerIDs := make([]uint64, len(stringIDs))
-	for i, id := range stringIDs {
-		var intID int
-		intID, err := strconv.Atoi(id)
-		if err != nil {
-			fmt.Printf("Error updating token manager from database: failed converting ID %s to int: %s", id, err)
-		}
-		integerIDs[i] = uint64(intID)
+func (tm *TokenManager) UpdateFromDatabase(logger *zap.SugaredLogger, db database.Database) {
+	tokens, err := db.GetTokenRecords()
+	if err != nil {
+		logger.Errorf("cannot fetch token to update from the database: %w", err)
```

Comment:
Should probably return after logging here; or return the error for logging at the call site.

Log message could be: `Failed to update JWT tokens from database: %v`

Reply:
Fixed.